### PR TITLE
Introduce ies-route53 cookbook

### PR DESCRIPTION
**Pull request by baccenfutter**
Issue: easybib/issues#5085

---- 

This PR introduces an `ies-route53` wrapper cookbook.

---- 

Todos:
- Extend node-attributes by `ies-route53` block on all production stacks *(see: all playgrounds)*
 - [x] API
 - [x] CitationAPI
 - [x] Easybib
 - [ ] Fruitkid
 - [x] IEScholar
 - [x] InfoLit
 - [x] Ops Infrasturucture
 - [x] QA
 - [ ] Research.easybib.com
 - [ ] Services

### CODE REVIEW

**Comment by baccenfutter** [comment]
`nginxphpapp2.easybib_playground.aws.ezbib.com` should boot with an `ies-route::add` and verify a test-run. If not, you might explicitly have to execute that recipe.

I still have to adjust the node-attributes of all stacks (see: https://github.com/easybib/issues/issues/5085#issuecomment-152472575).

I am assigning this to you so you can take a look where this is going, so far. Please assign back to be so I can continue to take care of the above issue.

File: ies-route53/recipes/add.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'net/http'
+Chef::Resource.send(:include, EasyBib)
+
+include_recipe 'route53'
+
+route53_record 'create a record' do
+  host_name = get_hostname(node, true)
```
Comment (till) [inline ies-route53/recipes/add.rb]: I would probably do these assignments outside the `route53_record` provider.

File: ies-route53/recipes/add.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  value Net::HTTP.get(URI.parse('http://169.254.169.254/latest/meta-data/public-ipv4'))
+  type  'A'
+  ttl                   node['ies-route53']['zone']['ttl']
+  zone_id               node['ies-route53']['zone']['id']
+  aws_access_key_id     node['ies-route53']['zone']['custom_access_key']
+  aws_secret_access_key node['ies-route53']['zone']['custom_secret_key']
+  overwrite true
+  action :create
+  not_if do
+    node.fetch('ies-route53', {}).fetch('zone', {}).fetch('id', {}).nil?
```
Comment (till) [inline ies-route53/recipes/add.rb]: add an `is_aws` in there as well?

File: ies-route53/recipes/add.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+Chef::Resource.send(:include, EasyBib)
+
+include_recipe 'route53'
+
+route53_record 'create a record' do
+  host_name = get_hostname(node, true)
+  stack_name = get_normalized_cluster_name(node)
+  zone_name = node['ies-route53']['zone']['name']
+  name "#{host_name}.#{stack_name}.#{zone_name}"
+  value Net::HTTP.get(URI.parse('http://169.254.169.254/latest/meta-data/public-ipv4'))
```
Comment (till) [inline ies-route53/recipes/add.rb]: And this should be in a `node[]` attribute — `node['opsworks']['instance']['public_ip']` or something?

File: ies-route53/recipes/add.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+require 'net/http'
+Chef::Resource.send(:include, EasyBib)
+
+include_recipe 'route53'
+
+host_name = get_hostname(node, true)
+stack_name = get_normalized_cluster_name(node)
+zone_name = node['ies-route53']['zone']['name']
+region_id = Net::HTTP.get(URI.parse('http://169.254.169.254/latest/meta-data/placement/availability-zone'))
```
Comment (baccenfutter) [inline ies-route53/recipes/add.rb]: works like a charm...

```
# host nginxphpapp2.easybib_playground.us-east-1a.aws.ezbib.com
nginxphpapp2.easybib_playground.us-east-1a.aws.ezbib.com has address 107.21.155.0
```

Comment (baccenfutter) [inline ies-route53/recipes/add.rb]: Maybe not quite yet... There is still a tailing `a` for the availability zone within the region. This letter needs to be trimmed.

**Comment by baccenfutter** [comment]
@fh do you think that *(in regards to our cookbook refactoring issue)* we should `include_recipe` our `ies-route53::add` inside the `ies::role-generic`, rather than configuring it via OpsWorks Web-UI?

**Comment by fh** [comment]
@baccenfutter **never** add individual cookbooks/recipes to the web ui.

File: ies-route53/recipes/add.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+Chef::Resource.send(:include, EasyBib)
+
```
Comment (till) [inline ies-route53/recipes/add.rb]: Try: `node.force_override['route53']['fog_version'] = 1.33.0`

**Comment by baccenfutter** [comment]
This PR should be ready for master, now.

**Comment by till** [comment]
@baccenfutter let's merge this and trial it on the playgrounds for a few days!

**Pull request merged**